File: ios-swiftui/library/Sources/PlanetComponents/TspTooltip.swift
import SwiftUI

/// Simple overlay / popover tooltip.
public struct TspTooltip<Content: View>: View {
    public var text: String
    public var placement: String
    public var visible: Bool?
    public var theme: StarPlanetTheme
    public var content: Content

    @State private var open = false

    public init(
        text: String = "",
        placement: String = "top",
        visible: Bool? = nil,
        theme: StarPlanetTheme = .sky,
        @ViewBuilder content: () -> Content
    ) {
        self.text = text
        self.placement = placement
        self.visible = visible
        self.theme = theme
        self.content = content()
    }

    private var shown: Bool { visible ?? open }

    public var body: some View {
        let bubble = Group {
            if shown && !text.isEmpty {
                Text(text)
                    .font(.system(size: 12, weight: .bold))
                    .foregroundColor(.white)
                    .padding(.horizontal, 10)
                    .padding(.vertical, 6)
                    .background(theme.textPrimary)
                    .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
                    .shadow(color: theme.brandPrimary.opacity(0.18), radius: 8, y: 4)
                    .accessibilityAddTraits(.isStaticText)
            }
        }

        Group {
            switch placement {
            case "bottom":
                VStack(spacing: 4) { content; bubble }
            case "left":
                HStack(spacing: 4) { bubble; content }
            case "right":
                HStack(spacing: 4) { content; bubble }
            default:
                VStack(spacing: 4) { bubble; content }
            }
        }
        .onTapGesture {
            if visible == nil {
                open.toggle()
            }
        }
    }
}

File: flutter/library/lib/src/tsp_tooltip.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

/// Simple overlay tooltip — tap to toggle when [visible] is null.
class TspTooltip extends StatefulWidget {
  const TspTooltip({
    super.key,
    this.text = '',
    this.placement = 'top',
    this.visible,
    this.theme = StarPlanetTheme.sky,
    required this.child,
  });

  final String text;
  final String placement;
  final bool? visible;
  final StarPlanetTheme theme;
  final Widget child;

  @override
  State<TspTooltip> createState() => _TspTooltipState();
}

class _TspTooltipState extends State<TspTooltip> {
  bool _open = false;

  bool get _shown => widget.visible ?? _open;

  @override
  Widget build(BuildContext context) {
    final bubble = _shown && widget.text.isNotEmpty
        ? Container(
            margin: const EdgeInsets.all(4),
            padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 6),
            decoration: BoxDecoration(
              color: widget.theme.textPrimary,
              borderRadius: BorderRadius.circular(10),
              boxShadow: [
                BoxShadow(
                  color: widget.theme.brandPrimary.withValues(alpha: 0.18),
                  blurRadius: 10,
                  offset: const Offset(0, 4),
                ),
              ],
            ),
            child: Text(
              widget.text,
              style: const TextStyle(color: Colors.white, fontSize: 12, fontWeight: FontWeight.w700),
            ),
          )
        : null;

    Widget stack;
    switch (widget.placement) {
      case 'bottom':
        stack = Column(mainAxisSize: MainAxisSize.min, children: [widget.child, if (bubble != null) bubble]);
      case 'left':
        stack = Row(mainAxisSize: MainAxisSize.min, children: [if (bubble != null) bubble, widget.child]);
      case 'right':
        stack = Row(mainAxisSize: MainAxisSize.min, children: [widget.child, if (bubble != null) bubble]);
      case 'top':
      default:
        stack = Column(mainAxisSize: MainAxisSize.min, children: [if (bubble != null) bubble, widget.child]);
    }

    return GestureDetector(
      onTap: widget.visible == null ? () => setState(() => _open = !_open) : null,
      behavior: HitTestBehavior.opaque,
      child: stack,
    );
  }
}
